### ISSUE TITLE
OpenSSL 3.1 does not compile on NonStop because of mman dependency.

### ISSUE THREAD

**Issue by rsbeckerca** [openssl/openssl#19810]
The OpenSSL 3.1.0-alpha1 does not build on NonStop because of a reference to `mman.h`. This is a non-portable dependency.

```
#  include <sys/mman.h>
                         ^
"/home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.1_Monitor/apps/speed.c", line 66: catastrophic error(4093):
          cannot open source file "sys/mman.h"
```

There should be a mechanism to avoid this dependency as with 3.0.7.

**Comment by t8m** [comment]
Fix in #19811

**Comment by rsbeckerca** [comment]
LGTM
